google-java-format.gradle

```diff
@@ -0,0 +1,71 @@
+/**
+ * Reformat the diff from a given commit, formatBase. By default formatBase=HEAD.
+ * Requires that bash can run executable python 2.7 files directly.
+ *
+ * Usage:
+ * ./gradlew googleJavaFormat [-PformatBase=<format base>]
+ * 
+ * To enable this task when build, explicitly specify property "formatBase"
+ * For example: 
+ * ./gradlew build -PformatBase="HEAD~1"
+ * 
+ * Be aware that your IDE may lose editing history if the task modifies the code.
+ */
+task googleJavaFormat {
+    try {
+        exec {
+            executable = '/usr/bin/env'
+            args 'python2.7'
+        }
+    } catch (Exception e) {
+        print "python2.7 not found, will skip ${name}"
+        enabled = false
+    }
+}
+
+if (googleJavaFormat.enabled) {
+    def googleJavaFormatVer = '1.7'
+    def googleJavaFormatDir = "${rootDir}/buildscripts/google-java-format"
+    def googleJavaFormatJar
+
+    repositories {
```

Comment:
I'm concerned about "random" other files changing the repositories. I'd prefer this be in `build.gradle` itself.